R/grid_helpers.R

```diff
@@ -9,9 +9,23 @@ predict_model <- function(split, workflow, grid, metrics, submodels = NULL) {
   orig_rows <- as.integer(split, data = "assessment")
 
   if (length(orig_rows) != nrow(x_vals)) {
-    rlang::abort(paste0("Some assessment set rows are not available at ",
-                        "prediction time. Did your preprocessing steps ",
-                        "filter or remove rows?"))
+    msg <- paste0("Some assessment set rows are not available at ",
+                  "prediction time. ")
+
+    if (names(workflow$pre$actions) == "recipe") {
```

Comment:
You can use the `has_preprocessor_recipe()` helper here. I try to avoid digging into the workflow internals where possible, that way if they change we only have to update the helpers

Reply:
Ah, thanks for this reminder!